qiime2/sdk/tests/test_usage.py

```diff
@@ -231,6 +248,13 @@ def test_validated_missing_optional_value(self):
         ui.validate(self.signature)
         self.assertTrue(True)
 
+    def test_type_of_input(self):
+        test_inputs = usage.UsageInputs(x=[1, 2, 3], z={7, 8, 9})
+        test_inputs.validate(self.signature)
+
+        self.assertEqual(list, type(test_inputs.values['x']))
+        self.assertEqual(set, type(test_inputs.values['z']))
```

Comment:
Great! A slightly more idiomatic approach would use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIsInstance

```suggestion
        self.assertIsInstance(test_inputs.values['x'], list)
        self.assertIsInstance(test_inputs.values['z'], set)
```